### PR TITLE
More PipeWire cleanups

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
### Description

Various legibility and quality related cleanups.

### Motivation and Context

I dislike code that confuses me, and try to make it less confusing when I see it.

### How Has This Been Tested?

 - Run OBS in Linux / Wayland
 - Check that everything still works

### Types of changes

 - Code cleanup (non-breaking change which makes code smaller or more readable) 

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/linux-pipewire/pipewire.c [hunk 1]
```diff
@@ -1006,57 +1081,18 @@ void obs_pipewire_video_render(obs_pipewire *obs_pw, gs_effect_t *effect)
...
-		offset_y = 0;
-		break;
-	}
-	if (rot != 0) {
-		gs_matrix_push();
-		gs_matrix_rotaa4f(0.0f, 0.0f, 1.0f, RAD(rot));
-		gs_matrix_translate3f(-offset_x, -offset_y, 0.0f);
-	}
-	if (has_crop) {
+	flip = get_buffer_flip (obs_pw);
```
Comment (WizardCM) [inline plugins/linux-pipewire/pipewire.c:1086]:
```suggestion
	flip = get_buffer_flip(obs_pw);
```

File: plugins/linux-pipewire/pipewire.c [hunk 2]
```diff
@@ -1006,57 +1081,18 @@ void obs_pipewire_video_render(obs_pipewire *obs_pw, gs_effect_t *effect)
 	image = gs_effect_get_param_by_name(effect, "image");
 	gs_effect_set_texture(image, obs_pw->texture);
 
-	has_crop = has_effective_crop(obs_pw);
+	rotated = apply_rotation (obs_pw);
```
Comment (WizardCM) [inline plugins/linux-pipewire/pipewire.c:1084]:
```suggestion
	rotated = apply_rotation(obs_pw);
```

File: plugins/linux-pipewire/pipewire.c [hunk 3]
```diff
@@ -181,6 +181,76 @@ static inline bool has_effective_crop(obs_pipewire *obs_pw)
...
+	case SPA_META_TRANSFORMATION_90:
+	case SPA_META_TRANSFORMATION_180:
+	case SPA_META_TRANSFORMATION_270:
+		break;
+	}
+
+	return flip;
+}
+
+static bool apply_rotation(obs_pipewire *obs_pw)
```
Comment (kkartaltepe) [inline plugins/linux-pipewire/pipewire.c:207]: nit: push_rotation is probably more clear that this has side effects.

**Review by kkartaltepe — APPROVED** [review]

**Comment by columbarius** [comment]
LGTM

**Review by RytoEX — APPROVED** [review]
Looks fine at a glance.

**Pull request merged**